### PR TITLE
Feature/cax 1 prs@260 remove duplicated classes

### PR DESCRIPTION
**Pull request by algattik**
Removed duplicated classes for
- DTOs for event messages
- Mother classes to generate those DTOs for tests
- Misc utility classes (annotations for skipping coverage and enum validation)

### CODE REVIEW

File: coreservices/partsrelationshipservice/broker-proxy/src/main/java/net/catenax/brokerproxy/services/BrokerProxyService.java [hunk 1]
```diff
@@ -93,7 +91,7 @@ public void send(final PartRelationshipUpdateRequest updateRelationships) {
      */
     public void send(final PartAspectUpdateRequest updateAspect) {
         log.info("Sending PartAspectUpdate to broker");
-        final var message = PartAspectUpdateEvent.builder()
+        final var message = PartAspectUpdateRequest.builder()
```
Comment (cpeeyush) [inline coreservices/partsrelationshipservice/broker-proxy/src/main/java/net/catenax/brokerproxy/services/BrokerProxyService.java:94]: I think now we can remove this mapping as both classes are same. So we can publish directly.

File: coreservices/partsrelationshipservice/broker-proxy-integration-tests/src/test/java/net/catenax/brokerProxy/UpdatePartRelationshipTest.java [hunk 2]
```diff
@@ -87,18 +90,22 @@ public void updatedPartsAttributesWithNoEffectTime_failure() throws JsonProcessi
...
+    }
+
+    private PartRelationshipUpdateRequest getPartRelationshipUpdateRequest(Function<PartRelationshipUpdate.PartRelationshipUpdateBuilder, PartRelationshipUpdate.PartRelationshipUpdateBuilder> f) {
+        var updateRequest = generate.partRelationshipUpdateList();
+        List<PartRelationshipUpdate> relationships = updateRequest.getRelationships().stream().map(r -> f.apply(r.toBuilder()).build()).collect(Collectors.toList());
+        return updateRequest.toBuilder().withRelationships(relationships).build();
     }
 
     @Test
     public void updatedPartsAttributesWithNoStage_failure() throws JsonProcessingException {
```
Comment (cpeeyush) [inline coreservices/partsrelationshipservice/broker-proxy-integration-tests/src/test/java/net/catenax/brokerProxy/UpdatePartRelationshipTest.java:103]: As no checked exception thrown
```suggestion
    public void updatedPartsAttributesWithNoStage_failure() {
```

File: coreservices/partsrelationshipservice/broker-proxy-integration-tests/src/test/java/net/catenax/brokerProxy/UpdatePartAspectTest.java [hunk 3]
```diff
@@ -50,12 +48,12 @@ public void updatedPartAspectUpdateBadRequest_failure() {
...
 
     @Test
-    public void updatedPartAspectUpdateEmptyAspectList_failure() throws JsonProcessingException {
+    public void updatedPartAspectUpdateEmptyAspectList_failure() {
 
         var response =
             given()
                 .contentType(ContentType.JSON)
-                .body(brokerProxyMother.partAspectUpdateEmptyList())
+                .body(generate.partAspectUpdate().toBuilder().withAspects(null).build())
```
[the remark this answers is not in the document]
Comment (algattik) [reply coreservices/partsrelationshipservice/broker-proxy-integration-tests/src/test/java/net/catenax/brokerProxy/UpdatePartAspectTest.java:56]: I see, we should then do a ParameterizedTest with both null and empty inputs

File: coreservices/partsrelationshipservice/prs-api/src/test/java/net/catenax/prs/mappers/PartAspectUpdateEventToEntityMapperTests.java [hunk 4]
```diff
@@ -12,14 +12,14 @@
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class PartAspectUpdateEventToEntityMapperTests {
```
Comment (Izzzu) [inline coreservices/partsrelationshipservice/prs-api/src/test/java/net/catenax/prs/mappers/PartAspectUpdateEventToEntityMapperTests.java:14]: Class name should be renamed to PartAspectUpdateRequestToEntityMapper?

File: coreservices/partsrelationshipservice/prs-testing/src/main/java/net/catenax/prs/testing/EventMessageMother.java [hunk 5]
```diff
@@ -24,7 +27,7 @@
  * https://martinfowler.com/bliki/ObjectMother.html</a>
  */
 @SuppressWarnings("checkstyle:MagicNumber")
-public class RequestMother {
+public class EventMessageMother {
```
Comment (Izzzu) [inline coreservices/partsrelationshipservice/prs-testing/src/main/java/net/catenax/prs/testing/EventMessageMother.java:30]: Why renaming to EventMessageMother as it generates Requests?

File: coreservices/partsrelationshipservice/lombok.config [hunk 6]
```diff
@@ -1,3 +1,5 @@
 # Add @Generated annotation to generated code, so that it is skipped by JaCoCo
 lombok.addLombokGeneratedAnnotation = true
 
+# Add @SuppressFBWarnings annotation to generated code, so that it is skipped by SpotBugs
+lombok.extern.findbugs.addSuppressFBWarnings = true
```
Comment (cpeeyush) [inline coreservices/partsrelationshipservice/lombok.config:5]: is this add on purpose here with these changes ? Are we facing issue in some generated code ?

**Review by cpeeyush — APPROVED** [review]

File: coreservices/partsrelationshipservice/lombok.config (hunk 6 above)
Comment (algattik) [reply coreservices/partsrelationshipservice/lombok.config:5]: Yes there was a SpotBugs warning related to generated code (strange it didn't pop up before)

**Pull request merged**